modules/bibupload/lib/bibupload.py

```diff
@@ -674,7 +679,35 @@ def find_record_from_oaiid(oaiid):
         write_message("   Error during find_record_from_oaiid(): %s " % error,
                       verbose=1, stream=sys.stderr)
     if res:
-        return res[0][0]
+        if len(res) == 1:
+            return res[0][0]
+        else:
+            # Principle dupes detected.
+            # If the dupes are resolved only one record should have survived,
+            # and all others are deleted. => get non-deleted record(s)
+            lst = []
+            for r in res:
+                lst.append('recid:"%s"' % str(r[0]))
+            candidates = search_pattern(p=' or '.join(lst))
+            deleted    = search_pattern(p='980__c:"DELETED"')
```

Comment:
Note the defintion of deleted in Invenio is a bit larger: any subfield of 980 could in principle contain the word `DELETED` (at least that is what is assumed as convention in `bibupload`)